### ISSUE TITLE
Using link logistic changes results

### ISSUE THREAD

**Issue by acgrissom** [JohnLangford/vowpal_wabbit#835]
Hi,

We've been using VW, but we've noticed some strange behavior when using logistic regression with/without the --link logistic flag and OAA.  If I'm correct, this shouldn't change the results, but it does.
Using loss_function=logistic without link=logistic results in different results from the alternatives:

$ vw --loss_function=logistic -f synth.model --oaa 3 < train_synthetic_dataset.dat 
$ vw -i synth.model --loss_function logistic -t < test_synthetic_dataset.dat 
average loss = 0.816901

The results are the same regardless of whether I use link logistic in the testing phase.

However, if I use --link=logistic in the training phase:
$ vw -i synth.model --loss_function logistic -t < test_synthetic_dataset.dat 
average loss = 0.344064

I've attached the files we used for reference.  We've tested this in 8.0 and 8.0.2.

Thanks,
Alvin
